psiturk/dashboard/__init__.py

```diff
@@ -1,120 +1,133 @@
 
+
 def init_app(app):
     login_manager.init_app(app)
 
+
 class DashboardUser(UserMixin):
     def __init__(self, username=''):
         self.id = username
 
+
 @login_manager.user_loader
 def load_user(username):
     return DashboardUser(username=username)
 
+
 def login_required(view):
     @wraps(view)
     def wrapped_view(*args, **kwargs):
-        if app.login_manager._login_disabled: # for unit testing
+        if app.login_manager._login_disabled:  # for unit testing
             return view(*args, **kwargs)
         is_logged_in = current_user.get_id() is not None
         is_static_resource_call = str(request.endpoint) == 'dashboard.static'
         is_login_route = str(request.url_rule) == '/dashboard/login'
         if not (is_static_resource_call or is_login_route or is_logged_in):
             return login_manager.unauthorized()
         return view(*args, **kwargs)
+
     return wrapped_view
 
+
 def try_amt_services_wrapper(view):
     @wraps(view)
     def wrapped_view(**kwargs):
         try:
-            _ = services_manager.amt_services_wrapper # may throw error
-                                                      # if aws keys not set
+            _ = services_manager.amt_services_wrapper  # may throw error
+            # if aws keys not set
             if SESSION_SERVICES_MANAGER_MODE_KEY not in session:
                 app.logger.debug('setting session mode to {}'.format(services_manager.mode))
                 session[SESSION_SERVICES_MANAGER_MODE_KEY] = services_manager.mode
             else:
-                app.logger.debug('found session mode: {}'.format(session[SESSION_SERVICES_MANAGER_MODE_KEY]))
+                app.logger.debug(
+                    'found session mode: {}'.format(session[SESSION_SERVICES_MANAGER_MODE_KEY]))
                 services_manager.mode = session[SESSION_SERVICES_MANAGER_MODE_KEY]
                 app.logger.debug('I set services manager mode to {}'.format(services_manager.mode))
             return view(**kwargs)
         except Exception as e:
             message = e.message if hasattr(e, 'message') else str(e)
             flash(message, 'danger')
             return redirect(url_for('.index'))
+
     return wrapped_view
 
+
 @dashboard.before_request
 @login_required
 @try_amt_services_wrapper
 def before_request():
     pass
 
-@dashboard.route('/mode', methods=('GET','POST'))
+
+@dashboard.route('/mode', methods=('GET', 'POST'))
 def mode():
     if request.method == 'POST':
         mode = request.form['mode']
-        if mode not in ['live','sandbox']:
+        if mode not in ['live', 'sandbox']:
             flash('unrecognized mode: {}'.format(mode), 'danger')
         else:
             try:
                 services_manager.mode = mode
                 session[SESSION_SERVICES_MANAGER_MODE_KEY] = mode
                 flash('mode successfully updated to {}'.format(mode), 'success')
-            except:
-                flash(str(result.exception), 'danger')
```

Comment:
I have no idea what I was doing here with `result.exception`

Reply:
I noticed this this morning. I changed it to

```
except Exception as e:
    flash(str(e), 'danger')
```